modules/cactus-web/src/Calendar/Grid.tsx

```diff
@@ -0,0 +1,433 @@
+      let toFocus: HTMLElement | null = old
+      // First choice is the selected date (first one this month if there's multiple).
+      if (isSelected !== stubFalse) {
+        toFocus = grid.querySelector<HTMLElement>('[aria-selected="true"]:not(.outside-date)')
+      }
+      if (!toFocus) {
+        const fd = getFocusDate(year, month) || getFocusFromToday(year, month)
+        toFocus = queryDate(grid, fd) || grid.querySelector<HTMLElement>(INSIDE_DATE)
+      }
+      if (toFocus) {
+        if (old && old !== toFocus) {
+          old.tabIndex = -1
+        }
+        toFocus.tabIndex = 0
+      }
+    }
+  }, [isSelected, month, year]) // eslint-disable-line react-hooks/exhaustive-deps
+  const props: React.HTMLAttributes<HTMLDivElement> = rest
+  props.onFocus = (e) => {
+    e.target.tabIndex = -1
+    onFocus?.(e)
+  }
+  props.onBlur = (e) => {
+    // Set tabIndex so that focus will pick up where it left off when returning to the grid.
+    if (isFocusOut(e) && e.target.dataset.date) {
+      e.target.tabIndex = 0
+    }
+    onBlur?.(e)
+  }
+  props.onClick = (e) => {
+    onClick?.(e)
+    const target = e.target as HTMLElement
+    const selector = '.outside-date[aria-disabled="false"]'
+    if (!e.isDefaultPrevented() && target.matches(selector)) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const dateStr: string = target.dataset.date!
+      const date = new Date(...dateParts(dateStr))
+      setOverflow([dateStr, date])
+      onFocusOverflow?.(date, e)
+    }
+  }
+  props.onKeyDown = (e) => {
+    onKeyDown?.(e)
+    if (!e.isDefaultPrevented() && (e.target as HTMLElement).dataset.date) {
+      const overflowDate = onGridKeyDown(e)
+      if (overflowDate) {
+        setOverflow([toISODate(overflowDate), overflowDate])
+        onFocusOverflow?.(overflowDate, e)
+      }
+    }
+  }
+
+  const { labelDate, labelDisabled, weekdays } = labels
+  const header = React.useMemo(() => makeGridHeader(locale, weekdays), [locale, weekdays])
+  const grid = React.useMemo(
+    () => makeGrid(month, year, locale, isValidDate, labelDate, labelDisabled),
+    [month, year, locale, isValidDate, labelDate, labelDisabled]
+  )
+  return (
+    <div {...props} key={`${year}-${month}`} ref={gridRef} role="grid">
+      {header && (
+        <div role="row" key="header">
+          {header}
+        </div>
+      )}
+      {grid.map((row, ix) => (
+        <div role="row" key={ix}>
+          {row.map((dateProps, jx) => {
+            // The default `labelDate` function includes the weekday but we
+            // can't tell with a custom one, so add describedby just in case.
+            if (header && labelDate) {
+              dateProps['aria-describedby'] = header[jx].props.id
+            }
+            dateProps['aria-selected'] = isSelected(dateProps['data-date'])
+            return <span {...dateProps} key={jx} ref={setTabIndex} role="gridcell" />
```

Comment:
I noticed that the screen reader didn't give any indication that the cells were clickable/selectable.  I think we might want to do something like the following to change that.  I implemented this, and the behavior was better (although we'd need to tweak the CSS).

```suggestion
            return (
              <span key={jx} ref={setTabIndex} role="gridcell">
                <button {...dateProps} type="button" />
              </span>
            )
```

Reply:
Wasn't it like this before? It basically had `<button role="gridcell">` in the old calendar, and when I try that in the new one it only reads the grid instructions, not the button instructions.

Reply:
After testing the old DateInput, I found the "instructions" were given in the `aria-roledescription` which, as I said in the description, is not how that value is meant to be used. I'm going to check how screen readers interact with the built-in date input to see if there's anything I can glean from that, but for the most part I think it'd be better to make these improvements in a separate ticket.